src/src/Helpers/Networking.cpp

```diff
@@ -1682,6 +1682,125 @@ int http_authenticate(const String& logIdentifier,
+
+                                         for (int i = 0; i < keyCount; i++) // Use keyCount to limit the iteration
+                                         {
+                                           String key        = keys[i];
+                                           String value      = "";
+                                           int    startIndex = str.indexOf(key + "\":", startStringIndex);
+
+                                           if (startIndex == -1)
+                                           {
+                                             // Handle case where key is not found
+                                             value = "-256"; // Placeholder value
+                                           }
+                                           else
+                                           {
+                                             int endIndex = 0;
+
+                                             if (eventName != "current") { // in daily and hourly the values are stored in an
+                                               // array
+                                               // so
+                                               // get rid of the brackets and put the values in a csv
+                                               startIndex += key.length() + 3; // Move index past the key
+                                               endIndex    = str.indexOf("]", startIndex);
+                                             }
+                                             else {
+                                               startIndex += key.length() + 2; // Move index past the key
+                                               endIndex    = str.indexOf(",", startIndex);
+                                             }
+
+                                             // Find the index of the next comma
+                                             if ((endIndex == -1) || (endIndex > str.indexOf("}", startIndex)))
+                                             {
+                                               endIndex = str.indexOf("}", startIndex); // If no comma is found or comma comes after },
+                                                                                        // take
+                                                                                        // the
+                                                                                        // rest of the string
+                                             }
+
+                                             value = str.substring(startIndex, endIndex);
+                                             value.trim(); // Remove any surrounding whitespace
+                                           }
+
+                                           if (!csv.isEmpty())
+                                           {
+                                             csv += ",";

```

Comment:
```c++
csv += ',';
```
Will result in smaller code as you only need to deal with a char instead of calling a complete constructor of a String using a C-style string.